backend/controller/artefacts/artefact.go

```diff
@@ -0,0 +1,35 @@
+package artefacts
+
+import (
+	"context"
+	"github.com/TBD54566975/ftl/internal/sha256"
+	"io"
+)
+
+// Metadata container for an artefact's metadata
+type Metadata struct {
+	Executable bool
+	Path       string
+}
+
+// Artefact container for an artefact's payload and metadata
+type Artefact struct {
+	Digest   sha256.SHA256
+	Metadata Metadata
+	Content  []byte
+}
+
+type ArtefactKey struct {
+	Digest sha256.SHA256
+	// Temporary
+	id int64
```

Comment:
We shouldn't need this right? We can just look up by hash rather than ID.

Reply:
yep your right, now that the registry handles deployment / artifact linking and the id is found in query via digest we no longer need to store id here